src/server/scripts/World/npcs_special.cpp

```diff
@@ -2387,6 +2388,58 @@ class npc_stable_master : public CreatureScript
         }
 };
 
+enum TrainWreckerSpells
+{
+    GO_TOY_TRAIN          = 193963,
+    SPELL_TOY_TRAIN_PULSE =  61551,
+    SPELL_WRECK_TRAIN     =  62943,
+
+};
+class npc_train_wrecker : public CreatureScript
+{
+    public:
+        npc_train_wrecker() : CreatureScript("npc_train_wrecker") { }
+
+        struct npc_train_wreckerAI : public NullCreatureAI
+        {
+            npc_train_wreckerAI(Creature* creature) : NullCreatureAI(creature), _isChasing(false), _searchTimer(1 * IN_MILLISECONDS) { }
+            void Wreck(GameObject* target)
+            {
+                if (me->GetDistance(target) < 5.0f)
+                {
+                    me->SetFacingToObject(target);
+                    me->CastSpell(target, SPELL_WRECK_TRAIN, true);
+                    target->AI()->DoAction(1); // hack? is there a better way to hook on spell received on gameobject?
+                }
+                else
+                {
+                    me->StopMoving();
+                    float x, y, z;
+                    me->GetContactPoint(target, x, y, z, 3.5f);
+                    me->GetMotionMaster()->MovePoint(1, x, y, z);
+                    _isChasing = true;
+                }
+            }
+            void UpdateAI(uint32 diff) override
+            {
+                if (diff < _searchTimer)
+                    _searchTimer -= diff;
+                else
+                {
+                    if(!_isChasing)
+                        if (GameObject* target = me->FindNearestGameObject(GO_TOY_TRAIN, 40.0f))
+                            Wreck(target);
+                    _searchTimer = 3 * IN_MILLISECONDS;
```

Comment:
Oh. My bad.

P.S When can we start using http://en.cppreference.com/w/cpp/chrono/operator%22%22ms? Pretty please.